lib/mongo/auth/scram/conversation.rb

```diff
@@ -429,9 +429,15 @@ def xor(first, second)
           first.bytes.zip(second.bytes).map{ |(a,b)| (a ^ b).chr }.join('')
         end
 
+        def secure_compare(a, b)
```

Comment:
Using the word "secure" is too strong of a claim (the original attempt to implement this for cpython made the same claim and was rejected). compare_digest seems better.